buildSrc/subprojects/kotlin-dsl/src/main/kotlin/build/GradleApiParameterNamesTransform.kt

```diff
@@ -85,21 +89,37 @@ fun Project.withCompileOnlyGradleApiModulesWithParameterNames(vararg gradleModul
 }
 
 
+@AssociatedTransformAction(GradleApiParameterNamesTransformAction::class)
 internal
-class GradleApiParameterNamesTransform @Inject constructor(
-    private val publicApiIncludes: List<String>,
-    private val publicApiExcludes: List<String>
-) : ArtifactTransform() {
+interface GradleApiParameterNamesTransform {
```

Comment:
Is our recommendation to have `Transform` in the name of the transform action/parameter object? I think we should figure out if we recommend using verbs of nouny names with `Transform` as a suffix.